pyiceberg/table/snapshots.py

```diff
@@ -116,3 +144,202 @@ class MetadataLogEntry(IcebergBaseModel):
+        elif manifest.content == ManifestContent.DELETES:
+            self.added_delete_files += manifest.added_files_count or 0
+            self.removed_delete_files += manifest.deleted_files_count or 0
+        else:
+            raise ValueError(f"Unknown manifest file content: {manifest.content}")
+
+    def build(self) -> Dict[str, str]:
+        def set_non_zero(properties: Dict[str, str], num: int, property_name: str) -> None:
+            if num > 0:
+                properties[property_name] = str(num)
+
+        properties: Dict[str, str] = {}
+        set_non_zero(properties, self.added_size, 'added-files-size')
+        set_non_zero(properties, self.removed_size, 'removed-files-size')
+        set_non_zero(properties, self.added_files, 'added-data-files')
+        set_non_zero(properties, self.removed_files, 'removed-data-files')
+        set_non_zero(properties, self.added_eq_delete_files, 'added-equality-delete-files')
+        set_non_zero(properties, self.removed_eq_delete_files, 'removed-equality-delete-files')
+        set_non_zero(properties, self.added_pos_delete_files, 'added-position-delete-files')
+        set_non_zero(properties, self.removed_pos_delete_files, 'removed-position-delete-files')
+        set_non_zero(properties, self.added_delete_files, 'added-delete-files')
+        set_non_zero(properties, self.removed_delete_files, 'removed-delete-files')
+        set_non_zero(properties, self.added_records, 'added-records')
+        set_non_zero(properties, self.deleted_records, 'deleted-records')
+        set_non_zero(properties, self.added_pos_deletes, 'added-position-deletes')
+        set_non_zero(properties, self.removed_pos_deletes, 'removed-position-deletes')
+        set_non_zero(properties, self.added_eq_deletes, 'added-equality-deletes')
+        set_non_zero(properties, self.removed_eq_deletes, 'removed-equality-deletes')
+
+        return properties
+
+
+def truncate_table_summary(summary: Summary, previous_summary: Mapping[str, str]) -> Summary:
+    for prop in {
+        'total-data-files',
+        'total-delete-files',
+        'total-records',
+        'total-files-size',
+        'total-position-deletes',
+        'total-equality-deletes',
+    }:
+        summary[prop] = '0'
+
+    if value := previous_summary.get('total-data-files'):
+        summary['deleted-data-files'] = value
+    if value := previous_summary.get('total-delete-files'):
+        summary['removed-delete-files'] = value
+    if value := previous_summary.get('total-records'):
+        summary['deleted-records'] = value
+    if value := previous_summary.get('total-files-size'):
+        summary['removed-files-size'] = value
+    if value := previous_summary.get('total-position-deletes'):
+        summary['removed-position-deletes'] = value
+    if value := previous_summary.get('total-equality-deletes'):
+        summary['removed-equality-deletes'] = value
+
+    return summary
+
+
+def merge_snapshot_summaries(
```

Comment:
Because this modifies `Summary`, I think it should be internal.

Reply:
Great call!